script.js

```diff
@@ -1,4 +1,469 @@
+const makeDeck = () => {
+  let cardDeck = [];
+  let suits = ["♥️", "♦️", "♣️", "♠️"];
+  let rankIcons = [
+    "🅰️",
+    "2️⃣",
+    "3️⃣",
+    "4️⃣",
+    "5️⃣",
+    "6️⃣",
+    "7️⃣",
+    "8️⃣",
+    "9️⃣",
+    "🔟",
+    "🎃",
+    "👸🏻",
+    "🤴🏻",
+  ];
+
+  let suitIndex = 0;
+  while (suitIndex < suits.length) {
+    let currentSuit = suits[suitIndex];
+
+    let rankCounter = 0;
+    while (rankCounter <= 12) {
+      let cardName = rankIcons[rankCounter];
+      let blackjackValue = rankCounter + 1;
+      if (rankCounter === 0) {
+        blackjackValue = [1, 11];
+      } else if (
+        rankCounter === 10 ||
+        rankCounter === 11 ||
+        rankCounter === 12
+      ) {
+        blackjackValue = 10;
+      }
+
+      let card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        value: blackjackValue,
+      };
+
+      cardDeck.push(card);
+
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
 };
+
+// Program to shuffle cards given a particular deck
+const shuffleCards = (deck) => {
+  let i = 0;
+  while (i < deck.length) {
+    let randomIndex = getRandomIndex(deck.length);
+    let randomCard = deck[randomIndex];
+    let currentCard = deck[i];
+    deck[i] = randomCard;
+    deck[randomIndex] = currentCard;
+    i += 1;
+  }
+  return deck;
+};
+
+// Generate random index for shuffling
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+
+// Generate standard deck of 52 cards
+let deck = makeDeck();
+
+// Program to show cards
+const addCard = (card) => {
+  const div = document.getElementById("cards");
+  let p = document.createElement("p");
+  p.innerText = `${card.name} of ${card.suit}`;
+  div.appendChild(p);
+};
+
+// Program to create score
+const showScore = () => {
+  const score = document.getElementById("player-score");
+  score.innerText = `${playerScore}`;
+};
+
+// ==================== Global variables ====================
+let playerHand = [];
+let dealerHand = [];
+
+let gameMode = "start";
+let playerScore = 0;
+let dealerScore = 0;
+
+let playerState = "safe";
+let dealerState = "safe";
+
+// ==================== Draw card functions ====================
+// Draw 2 cards each for player and dealer
+const drawHand = (deck) => {
+  let card1 = deck.pop();
+  playerHand.push(card1);
+  addCard(card1);
+
+  dealerHand.push(deck.pop());
+
+  let card2 = deck.pop();
+  playerHand.push(card2);
+  addCard(card2);
+
+  dealerHand.push(deck.pop());
+};
+
+// Update player score by looping through the hand and capturing total value
+const updateScore = (hand, person) => {
+  let score = 0;
+  for (const card of hand) {
+    if (card.name === "🅰️") {
+      score += card.value[1];
+    } else {
+      score += card.value;
+    }
+  }
+  if (person === "player") {
+    playerScore = score;
+  } else {
+    dealerScore = score;
+  }
+};
+
+// Assess no. of aces and adjust scores accordingly
+const hasAce = (hand, person) => {
```

Comment:
You were asking if you handled the aces correctly. I would say there is no right or wrong. Everyone has their own way of handling it. Your method is one of the thousands of ways to do it(: